enhancements/machine-config/mco-state-reporting.md

```diff
@@ -0,0 +1,519 @@
+```
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARING   UPDATEINPROGRESS   UPDATEPOSTACTION   UPDATECOMPLETING   RESUMING
+ip-10-0-12-194.ec2.internal   False     False             False              False              True               False
+ip-10-0-17-102.ec2.internal   True      False             False              False              False              False
+ip-10-0-2-232.ec2.internal    True      False             False              False              False              False
+ip-10-0-59-251.ec2.internal   True      False             False              False              False              False
+ip-10-0-59-56.ec2.internal    True      False             False              False              False              False
+ip-10-0-6-214.ec2.internal    True      False             False              False              False              False
+```
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARING   UPDATEINPROGRESS   UPDATEPOSTACTION   UPDATECOMPLETING   RESUMING
+ip-10-0-12-194.ec2.internal   False     False             False               False             False              True
+ip-10-0-17-102.ec2.internal   True      False             False              False              False              False
+ip-10-0-2-232.ec2.internal    True      False             False              False              False              False
+ip-10-0-59-251.ec2.internal   True      False             False              False              False              False
+ip-10-0-59-56.ec2.internal    True      False             False              False              False              False
+ip-10-0-6-214.ec2.internal    True      False             False              False              False              False
+```
+
+
+and if you wanted to look at oc describe for more verbose details of previous events that happened on the node, you can do so using `oc describe machineconfignodes/<name>`
+(output below is not correlated to the above example)
+
+```console
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               Node has completed update to config rendered-master-03c160a7e646004356a0b038dd22a97a
+    Reason:                NodeUpgradeComplete
+    Status:                True
+    Type:                  Updated
+    Last Transition Time:  2023-10-17T13:05:02Z
+    Message:               Detected Diff in MachineConfigs rendered-master-03c160a7e646004356a0b038dd22a97a and rendered-master-03c160a7e646004356a0b038dd22a97a
+    Reason:                NodeUpgradePreparing
+    Status:                False
+    Type:                  UpdatePreparing
+	Last Transition Time:  2023-10-17T13:06:32Z
+    Message:               Applying files to node. Using new Ignition files:...
+    Reason:                UpgradeInProgressApplyingFilesAndOSChanges
+    Status:                False
```

Comment:
See Above. Status is false so this represents a condition that previously occurred.

If we do not have the False conditions, this whole structure loses its value. Keeping the previous messages from when the conditions were true allows users to see what happened. Otherwise, there is no history if the upgrade moves quickly